lib/compiler.js

```diff
@@ -499,14 +517,17 @@ function buildMiddlewareInstructions(rootDir, config) {
   };
 }
 
-function resolveMiddlewarePath(rootDir, middleware) {
+function resolveMiddlewarePath(rootDir, middleware, config) {
   var resolved = {};
 
   var segments = middleware.split('#');
   var pathName = segments[0];
   var fragment = segments[1];
   var middlewarePath = pathName;
-  var opts = { strict: true };
+  var opts = {
+    strict: true,
+    optional: config.optional
```

Comment:
`config.optional` is a string, `opts.optional` is boolean. Perhaps it's better to coerce the value here?

```js
optional: !!config.optional
```